### ISSUE TITLE
Visual Studio 17.4 Preview 4+ breaks Add New File

### ISSUE THREAD

**Issue by Stekeblad** [madskristensen/AddAnyFile#135]
**Describe the bug**
The latest previews of Visual Studio 17.4 introduced a add new file dialog similar to the one by this extension and assigned it to the same keybinding (shift+F2). This resulted for me in that Visual Studio's own keybinding takes priority.

I have tried rebinding the keys for the dialog added to Visual Studio and then reinstalling the extension but can't get the Add New File dialog to show anymore.

I have tried looking for a keybinding registered by Add New File but can't find anything so I can't assign it a new either.

**To Reproduce**
1. Install a preview version of Visual Studio from before the conflicting dialog was introduced, for example 17.4 Preview 1
2. Install Add New File from the extensions menu in Visual Studio.
3. Restart and check that Add New File works as intended
    a. Open a project
    b. Select a location in Solution explorer and press shift+F2
    c. Create a new .cs file
    d. Add New File adds a new file outlining an empty class
4. Update Visual Studio to the latest 17.4 preview
5. Open the same project as earlier for a new test
    a. Select a location in Solution explorer and press shift+F2
    b. Create a new .cs file, note that the dialog looks different.
    c. Visual Studio creates a new, completely empty file.

**Expected behavior**
Even after Visual Studio has added its own similar add new file dialog it should still be possible to use this extension and it's templates.

**Comment by madskristensen** [comment]
Why do you want to use this extension over the new built in feature?

**Comment by Stekeblad** [comment]
I love how this extension quickly creates the right type of file with some basic code (like a empty c# class) by just telling it filename, location and extension.

I have to admit I have not looked into if there any options for the new feature but with the default settings it creates an empty file when doing the same steps as with Add New File.

To not get an empty file it looks like you need to click a button in the pop-up that opens the add new item window with all templates and you have to look for the right template. That is no big difference from the standard Visual Studio way before.

Additionally, if it has not improved since before I started using Add New File, so feels the Add New Item dialog slow and the template I wanted to use rarely was quickly accessible, often being under a different tab or having to scroll down in the template list.

While I think the extensions template selector is a bit limited in what you can do (*controller.cs is special cased in code but other suffixes can't have templates without modifying the extension code) this extension has a another benefit for being easy to create custom templates without having to create a special package or extension for them.

**Comment by ryanbuening** [comment]
Same thoughts as @Stekeblad. It is unfortunate that we can't disable the new Add Quick File feature and continue to use this extension in VS Preview. I tried removing the new `AddQuickFile` shortcut (Shift + F2) in the options, but doing that does not allow for the extension prompt to show.

**Comment by tuliopaim** [comment]
You can go to <b>Options -> Environment -> Keyboard</b>, remove the shortcut of "AddQuickFile", and assign "Shift+ F2" to "NewEmptyFile"

**Comment by JJ-Chen** [comment]
Thanks @tuliopaim , it works! Save my day!

As visual studio in "What's New?" says: applying templates will be support in Preview 2.
We could look forward to the next version.
![image](https://user-images.githubusercontent.com/26740614/200732740-42dbb9a4-7ee7-452d-8015-89aa53559d9f.png)

But so far, this extension still the most efficient way to add a new file.🧐

**Comment by marekpetak** [comment]
VS version 17.4.0 (non preview) has this as well. VS will only create an empty file. Nowhere near as good as the extension at them moment!

**Comment by chucker** [comment]
> Why do you want to use this extension over the new built in feature?

One reason I ran into today: in your extension, I can write `../../Roles/MyNewType.cs`. It'll navigate upwards, auto-generate the folder `Roles`, and generate the file.

VS's feature, meanwhile, will tell me (as of 17.5.0 Preview 1.0) that this contains illegal characters.

**Comment by ivnext** [comment]
> Why do you want to use this extension over the new built in feature?

It is broken, but your original works brilliantly. This one creates an empty file (no content). I use this numerous times per-day and love it. Until Microsoft fixes this new built-in feature, I am going with the solution by @tuliopaim above. Thanks again for all of your awesome work!

**Comment by mcb2001** [comment]
Anyone found a solution for this? I really miss the SHIFT + F2 feature in my workflow - the new empty files are driving me crazy

**Comment by vpekarek** [comment]
Check the @tuliopaim solution:

_You can go to Options -> Environment -> Keyboard, remove the shortcut of "AddQuickFile", and assign "Shift+ F2" to "NewEmptyFile"_

**Comment by bplus0** [comment]
The solution @tuliopaim suggested is still working as of 11/28. Losing this extension derailed my afternoon. So happy I found this thread!

**Comment by sgf** [comment]
how about add new shortcut likes "Shift+F3"

**Comment by StevenTCramer** [comment]
> 

Reading this thread, I wonder why the VS feature isn't the same?  W

> Why do you want to use this extension over the new built in feature?

I can not even find docs on AddQuickFile?  So how would I even know how to use the feature?

**Comment by bplus0** [comment]
> > Why do you want to use this extension over the new built in feature?
> 
> I can not even find docs on AddQuickFile? So how would I even know how to use the feature?

I can't comment on the docs. I don't know where they are. But the reason I like the extension more is AddQuickFile will create a code behind file if you're using razor pages (most of the time) and scaffold c# classes with the namespace and proper file structure.

Visual Studio built in just does a blank file with the proper extension.

**Comment by StevenTCramer** [comment]
> > > Why do you want to use this extension over the new built in feature?
> > 
> > 
> > I can not even find docs on AddQuickFile? So how would I even know how to use the feature?
> 
> I can't comment on the docs. I don't know where they are. But the reason I like the extension more is AddQuickFile will create a code behind file if you're using razor pages (most of the time) and scaffold c# classes with the namespace and proper file structure.
> 
> Visual Studio built in just does a blank file with the proper extension.

Yes, the real question, IMO, is why bother adding the AddQuickFile  feature at all if not going to do it correctly.  This extension is hugely popular and probably the motivation so why not implement it and if not why stomp on the hot key?
